hierarchyAPI/get_node_hierarchy_test.go

```diff
@@ -0,0 +1,94 @@
+package hierarchyAPI
+
+import (
+	"fmt"
+	"github.com/ONSdigital/dp-api-tests/testDataSetup/neo4j"
+	"github.com/ONSdigital/go-ns/log"
+	"github.com/gavv/httpexpect"
+	"github.com/satori/go.uuid"
+	. "github.com/smartystreets/goconvey/convey"
+	"net/http"
+	"os"
+	"testing"
+)
+
+func TestSuccessfullyGetNodeHierarchy(t *testing.T) {
+	instanceID := uuid.NewV4().String()
+	cpiCode := "cpi1dim1T120000"
+	hierarchyAPI := httpexpect.New(t, cfg.HierarchyAPIURL)
+
+	Convey("Given an existing hierarchy", t, func() {
+
+		datastore := neo4j.NewDatastore(cfg.Neo4jAddr, instanceID, neo4j.HierarchyTestData)
+		Convey("When a root hierarchy node is requested", func() {
+
+			err := datastore.Setup()
+			if err != nil {
+				log.ErrorC("Unable to setup test data", err, nil)
+				os.Exit(1)
+			}
+
+			Convey("Then a child hierarchy node is return as a response", func() {
+				response := hierarchyAPI.GET("/hierarchies/{instance_id}/{dimension_name}/{code}", instanceID, "aggregate", cpiCode).
+					Expect().Status(http.StatusOK).JSON().Object()
+
+				// Check root node
+				response.Value("has_data").Boolean().Equal(true)
+				response.Value("no_of_children").Number().Equal(6)
+				selfLink := response.Value("links").Object().Value("self").Object()
+				codeLink := response.Value("links").Object().Value("code").Object()
+				codeLink.Value("href").String().
+					Equal("http://localhost:22400/code-list/e44de4c4-d39e-4e2f-942b-3ca10584d078/code/cpi1dim1T120000")
+				selfLink.Value("href").String().
+					Equal(fmt.Sprintf("http://localhost:22600/hierarchies/%s/aggregate/%s", instanceID, cpiCode))
+
+				// Check first child node
+				first := response.Value("children").Array().First().Object()
+				first.Value("has_data").Boolean().True()
+				first.Value("label").String().Equal("12.1 Personal care")
+				first.Value("no_of_children").Number().Equal(2)
+				firstSelfLink := first.Value("links").Object().Value("self").Object()
+				firstCodeLink := first.Value("links").Object().Value("code").Object()
+				firstCodeLink.Value("href").String().
+					Equal("http://localhost:22400/code-list/e44de4c4-d39e-4e2f-942b-3ca10584d078/code/cpi1dim1G120100")
+				firstSelfLink.Value("href").String().
+					Equal(fmt.Sprintf("http://localhost:22600/hierarchies/%s/aggregate/cpi1dim1G120100", instanceID))
+			})
+		})
+
+		err := datastore.TeardownHierarchy()
+		if err != nil {
+			log.ErrorC("Unable to tear down test data", err, nil)
+			os.Exit(1)
+		}
+	})
+}
+
+func TestErrorStatesGetNodeHierarchy(t *testing.T) {
```

Comment:
I know you have followed the spec, but if an instance or dimension against an instance is not found is this not a bad request when trying to find hierarchy node?

Either way there are two tests missing, one to return error when `instance_id` is wrong (not found -> resulting in a bad request) and `dimension_name` is wrong (not found -> resulting in a bad request).

Reply:
I added the additional tests, you have requested.